pkg/addons/seaweedfs/seaweedfs.go

```diff
@@ -0,0 +1,208 @@
+			"type": "persistentVolumeClaim",
+			"size": "1Gi",
+		},
+		"disableHttp": true,
+	},
+	"volume": map[string]interface{}{
+		"replicas": 1,
+		"dataDirs": []map[string]interface{}{
+			{
+				"name":         "data1",
+				"type":         "persistentVolumeClaim",
+				"size":         "10Gi",
+				"storageClass": "openebs-hostpath",
+				"maxVolumes":   0,
+			},
+		},
+	},
+	"filer": map[string]interface{}{
+		"data": map[string]interface{}{
+			"type": "persistentVolumeClaim",
+			"size": "1Gi",
+		},
+		"logs": map[string]interface{}{
+			"type": "persistentVolumeClaim",
+			"size": "1Gi",
+		},
+		"s3": map[string]interface{}{
+			"enabled":              true,
+			"enableAuth":           true,
+			"existingConfigSecret": "seaweedfs-s3-access-secret",
+		},
+	},
+}
+
+var (
+	rwAccessKey = helpers.RandString(20)
+	rwSecretKey = helpers.RandString(40)
+	roAccessKey = helpers.RandString(20)
+	roSecretKey = helpers.RandString(40)
+)
+
+// SeaweedFS manages the installation of the SeaweedFS helm chart.
+type SeaweedFS struct {
+	isAirgap bool
+}
+
+// Version returns the version of the SeaweedFS chart.
+func (o *SeaweedFS) Version() (map[string]string, error) {
+	return map[string]string{"SeaweedFS": "v" + Version}, nil
+}
+
+func (a *SeaweedFS) Name() string {
+	return "SeaweedFS"
+}
+
+// HostPreflights returns the host preflight objects found inside the SeaweedFS
+// Helm Chart, this is empty as there is no host preflight on there.
+func (o *SeaweedFS) HostPreflights() (*v1beta2.HostPreflightSpec, error) {
+	return nil, nil
+}
+
+// GetProtectedFields returns the protected fields for the embedded charts.
+// placeholder for now.
+func (o *SeaweedFS) GetProtectedFields() map[string][]string {
+	protectedFields := []string{}
+	return map[string][]string{releaseName: protectedFields}
+}
+
+// GenerateHelmConfig generates the helm config for the SeaweedFS chart.
+func (o *SeaweedFS) GenerateHelmConfig(onlyDefaults bool) ([]v1beta1.Chart, []v1beta1.Repository, error) {
+	if !o.isAirgap {
+		return nil, nil, nil
+	}
+
+	chartConfig := v1beta1.Chart{
+		Name:      releaseName,
+		ChartName: ChartName,
+		Version:   Version,
+		TargetNS:  namespace,
+		Order:     2,
+	}
+
+	repositoryConfig := v1beta1.Repository{
+		Name: "seaweedfs",
+		URL:  ChartURL,
+	}
+
+	valuesStringData, err := yaml.Marshal(helmValues)
+	if err != nil {
+		return nil, nil, fmt.Errorf("unable to marshal helm values: %w", err)
+	}
+	chartConfig.Values = string(valuesStringData)
+
+	return []v1beta1.Chart{chartConfig}, []v1beta1.Repository{repositoryConfig}, nil
+}
+
+// Outro is executed after the cluster deployment.
+func (o *SeaweedFS) Outro(ctx context.Context, cli client.Client) error {
+	if !o.isAirgap {
+		return nil
+	}
+
+	loading := spinner.Start()
+	loading.Infof("Waiting for SeaweedFS to be ready")
+	if err := kubeutils.WaitForNamespace(ctx, cli, namespace); err != nil {
+		loading.Close()
+		return err
+	}
+	loading.Infof("SeaweedFS namespace is ready")
```

Comment:
How would SeaweedFS become ready without the s3 secret below? I remember it crashes if it cannot find it on startup.

Reply:
Oh this is the namespace, never mind!